include/aligator/core/traj-opt-problem.hxx

```diff
@@ -17,11 +17,11 @@ TrajOptProblemTpl<Scalar>::TrajOptProblemTpl(
     xyz::polymorphic<UnaryFunction> init_constraint,
     const std::vector<xyz::polymorphic<StageModel>> &stages,
     xyz::polymorphic<CostAbstract> term_cost)
-    : init_condition_(init_constraint), stages_(stages), term_cost_(term_cost),
-      unone_(term_cost->nu) {
+    : init_constraint_(std::move(init_constraint)), stages_(stages),
+      term_cost_(std::move(term_cost)), unone_(term_cost_->nu) {
```

Comment:
Do we really gain anything by using std::move here? We may be avoiding a copy, but does this really matter? (Just wondering, I'm curious)

Reply:
Since this is not a real-time critical path it's not so important, but it's usually good practice to call std::move() when taking complex-typed arguments (especially ones where the copy constructor allocates) by value